tensorrt_llm/_torch/auto_deploy/transform/library/attention.py

```diff
@@ -414,91 +325,242 @@ def _grouped_attn_replacement_4(q, k, v, attn_mask, dropout_p, scale):
     )
 
 
-def match_attention_layout(gm: GraphModule, attention_op: Type[AttentionDescriptor]) -> None:
+@TransformRegistry.register("match_repeat_kv")
+class MatchRepeatKV(BaseTransform):
+    """
+    Match and replace the repeat_kv pattern with torch.ops.auto_deploy.torch_attention_repeat_kv.
     """
-    Match and transform attention operations to match the layout expected by the attention backend.
 
-    If the attention backend expects 'bnsd' layout (batch, num_heads, seq_len, head_dim), which
-    is the default for SDPA operations, we don't need to transform anything.
+    def _apply(
+        self, gm: GraphModule, cm: CachedSequenceInterface, factory: ModelFactory
+    ) -> Tuple[GraphModule, TransformInfo]:
+        def register_repeat_kv(patterns: ADPatternMatcherPass):
+            dummy_args = [
+                torch.randn(8, 8, 16, 64, device="cuda", dtype=torch.float16),
+                7,
+            ]
+            register_ad_pattern(
+                search_fn=_repeat_kv_pattern,
+                replace_fn=_repeat_kv_repl,
+                patterns=patterns,
+                dummy_args=dummy_args,
+                op_ignore_types={
+                    torch.ops.aten.reshape.default: (int,),
+                    torch.ops.aten.expand.default: (int,),
+                },
+                scalar_workaround={"n_rep": dummy_args[1]},
+            )
 
-    If the backend expects 'bsnd' layout (batch, seq_len, num_heads, head_dim), we insert
-    appropriate transposes before and after SDPA operations and replace them with bsnd_grouped_sdpa.
+        num_kv_patterns = _apply_pattern(gm, "Repeat KV", register_repeat_kv, shape_prop=True)
+
+        info = TransformInfo(
+            skipped=False,
+            num_matches=num_kv_patterns,
+            is_clean=True,
+            has_valid_shapes=True,
```

Comment:
```suggestion
            is_clean=False,
            has_valid_shapes=False,
```

when just relying on the mechanism to configure it via the `run_shape_prop` field